File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int size = nums.size();
        int l=0;
        int r = size-1;
        int pos = size-1;
        vector<int>ans(size);
        while(r>=l)
        {
           if(abs(nums[l])<abs(nums[r]))
                {
                    ans[pos--]=nums[r]*nums[r];
                    r--;
                }
            else{
                ans[pos--]=nums[l]*nums[l];
                l++;
            }
        }
        return ans;
    }
};

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        ans=[None]*len(nums)
        l=0
        r=len(nums)-1
        k=r
        while(l<=r):
            lsq=nums[l]*nums[l]
            rsq=nums[r]*nums[r]
            if(lsq>rsq):
                ans[k]=lsq
                l+=1
            else:
                ans[k]=rsq
                r-=1

            k-=1
        return ans
        